pkg/operator/ceph/cluster/watcher.go

```diff
@@ -130,6 +144,319 @@ func (c *clientCluster) onK8sNode(ctx context.Context, object runtime.Object) bo
+}
+
+func (c *clientCluster) fennceCephFSVolumes(
+	ctx context.Context, node corev1.Node, cluster *cephv1.CephCluster,
+	clusterInfo *cephclient.ClusterInfo, listCephFSPV []corev1.PersistentVolume) error {
+
+	for _, pv := range listCephFSPV {
+		logger.Infof("cephfs PV NAME %v", pv.Spec.CSI.VolumeAttributes)
+
+		status, err := cephclient.StatusWithUser(c.context, clusterInfo)
+		if err != nil {
+			return fmt.Errorf("failed to get ceph status for check active mds. %v", err)
+		}
+
+		var activeMDS string
+		for _, fsRank := range status.Fsmap.ByRank {
+			if fsRank.Status == "up:active" {
+				activeMDS = fsRank.Name
+			}
+		}
+
+		args := []string{"tell", fmt.Sprintf("mds.%s", activeMDS), "client", "ls", "--format", "json"}
+		cmd := cephclient.NewCephCommand(c.context, clusterInfo, args)
+		cmd.JsonOutput = true
+
+		buf, err := cmd.Run()
+		if err != nil {
+			return fmt.Errorf("failed to list watchers for cephfs pool/subvoumeName %s/%s. %v", pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["subvolumeName"], err)
+		}
+		ips, err := cephFSMDSClientMarshal(buf, pv)
+		if err != nil || ips == nil {
+			return fmt.Errorf("failed to unmarshal cephfs mds  output. %v", err)
+		}
+
+		err = c.createNetworkFence(ctx, pv, node, cluster, ips)
+		if err != nil {
+			return fmt.Errorf("failed to create network fence for node %q. %v", node.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func cephFSMDSClientMarshal(output []byte, cephFSPV corev1.PersistentVolume) ([]string, error) {
+	type entity struct {
+		Addr struct {
+			Addr  string `json:"addr"`
+			Nonce int    `json:"nonce"`
+		} `json:"addr"`
+	}
+
+	type clientMetadata struct {
+		Root string `json:"root"`
+	}
+
+	type cephFSData struct {
+		Entity         entity         `json:"entity"`
+		ClientMetadata clientMetadata `json:"client_metadata"`
+	}
+
+	var data []cephFSData
+	err := json.Unmarshal(output, &data)
+	if err != nil {
+		return []string{}, err
+	}
+
+	for _, d := range data {
+		if cephFSPV.Spec.CSI.VolumeAttributes["subvolumePath"] == d.ClientMetadata.Root {
+			logger.Infof("cephfs mds client ips to fence %v", d.Entity.Addr)
+			return []string{d.Entity.Addr.Addr + "/32"}, nil
+		}
+	}
+
+	return []string{}, nil
+}
+
+func (c *clientCluster) createNetworkFence(ctx context.Context, pv corev1.PersistentVolume, node corev1.Node, cluster *cephv1.CephCluster, cidr []string) error {
+	logger.Warningf("Blocking node IP %s", cidr)
+
+	networkFence := &addonsv1alpha1.NetworkFence{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      node.Name,
+			Namespace: cluster.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(cluster, cephv1.SchemeGroupVersion.WithKind("CephCluster")),
+			},
+		},
+		Spec: addonsv1alpha1.NetworkFenceSpec{
+			Driver:     pv.Spec.CSI.Driver,
+			FenceState: addonsv1alpha1.Fenced,
+			Secret: addonsv1alpha1.SecretSpec{
+				Name:      pv.Annotations["volume.kubernetes.io/provisioner-deletion-secret-name"],
```

Comment:
Please double-check this one. The PVC's Created by the older version of CephCSI/Rook will not have the annotation. In that case, you need to read it from the storageclass.

Reply:
not sure where to check I tried to look at Kubernetes release notes and all but no luck.

Reply:
You can just install older version of Rook i.e 1.8 or 1.9 and check

Reply:
This still need to be addressed.

Reply:
right, I need to test this.

Reply:
annotation was not present, so now fetching from storage class